### PR TITLE
Migrate `ballerina/ldap` module

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
This PR migrate the implementation of `ballerina/ldap` module with the Git history.

Related to https://github.com/ballerina-platform/ballerina-lang/issues/25050

### CODE REVIEW

File: gradle/javaProject.gradle [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
```
Comment (BuddhiWathsala) [inline gradle/javaProject.gradle]: Do we need this configuration file?

File: ldap-ballerina/build.gradle [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+import org.apache.tools.ant.taskdefs.condition.Os
+
+description = 'Ballerina - LDAP Ballerina'
+
+configurations {
+    jbalTools
+    ldapBallerinaBirDir
+    ldapBallerinaJar
```
Comment (BuddhiWathsala) [inline ldap-ballerina/build.gradle]:
```suggestion
```

File: ldap-ballerina/build.gradle [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    externalJars
+}
+
+dependencies {
+    jbalTools ("org.ballerinalang:jballerina-tools:${ballerinaLangVersion}") {
+        transitive = false
+    }
+    compile project(':ldap-native')
+    externalJars (group: 'org.apache.directory.server', name: 'apacheds-all', version: '2.0.0-M24') {
```
Comment (BuddhiWathsala) [inline ldap-ballerina/build.gradle]: Shall we parameterize this version as `${ballerinaLangVersion}`?

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

**Comment by BuddhiWathsala** [comment]
@ldclakmal, don't we need this `apacheds-all-2.0.0-M24.jar` dependency anymore?

**Comment by ldclakmal** [comment]
@BuddhiWathsala No. It was used only for a Java test class. Since I removed that with docker image, it is no longer needed.

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**